Godeps/_workspace/src/k8s.io/kubernetes/pkg/apiserver/apiserver.go

```diff
@@ -177,7 +177,11 @@ func serviceErrorHandler(requestResolver *APIRequestInfoResolver, apiVersions []
 		}
 	}
 
-	errorJSON(apierrors.NewGenericServerResponse(serviceErr.Code, "", "", "", "", 0, false), codec, response.ResponseWriter)
+	resourceOrKind := requestInfo.Kind
+	if len(resourceOrKind) == 0 {
+		resourceOrKind = requestInfo.Resource
```

Comment:
I could change the `NewGenericServerResponse` to accept resource as well.  I'd have to change how it constructs the messages but if you think this approach is workable then I don't think that's a big deal.  My main concern was getting it to pass the info along with getting the RequestInfo parsed properly when there are api prefixes that are not a part of Kube